### PR TITLE
Refs [TASK][38397] Handle events SearchPlaceScreen

### PR DESCRIPTION
**Pull request by vinhmone**
## Related Tickets
- [#Task 38397: Handle events SearchPlaceScreen](https://edu-redmine.sun-asterisk.vn/issues/38397)
## WHAT
- 
## Review Checklist

Category | View Point | Description | Expected Reviewer Answer | Self review | Reviewer2 (name)
--- | --- | --- | --- | --- | ---
Conventions | Does the code follow Sun* coding style and coding conventions? | https://github.com/framgia/coding-standards/tree/master/eng/android | YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Redmine | Does the ticket follow Sun* Redmine working process?  | https://github.com/framgia/Training-Guideline/blob/master/WorkingProcess/redmine/redmine.md| YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Documentation | Is there any incomplete code? If so, should it be removed or flagged with a suitable marker like ‘TODO’? |  | YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

File: app/src/main/java/com/sun/findflight/ui/home/HomeFragment.kt [hunk 1]
```diff
@@ -115,9 +127,13 @@ class HomeFragment : BaseFragment<FragmentHomeBinding>(), View.OnClickListener {
...
         if (textView1.text.isNotEmpty() && textView2.text.isNotEmpty()) {
-            val temp = textView1.text
+            val tempName = textView1.text
             textView1.text = textView2.text
-            textView2.text = temp
+            textView2.text = tempName
+
+            val tempPlace = placeFromObject
+            placeFromObject = placeToObject
+            placeToObject = tempPlace
```
Comment (datdt-1935) [inline app/src/main/java/com/sun/findflight/ui/home/HomeFragment.kt]: dùng cái này được không em?
https://stackoverflow.com/a/45377921/13181463

File: app/src/main/java/com/sun/findflight/ui/main/MainActivity.kt [hunk 2]
```diff
@@ -42,7 +54,34 @@ class MainActivity : BaseActivity<ActivityMainBinding>(), View.OnClickListener {
...
+        else supportFragmentManager.popBackStack()
+    }
+
+    fun enableBackButton() {
+        supportActionBar?.setDisplayHomeAsUpEnabled(true)
+    }
+
+    fun disableBackButton() {
+        supportActionBar?.setDisplayHomeAsUpEnabled(false)
+    }
```
Comment (datdt-1935) [inline app/src/main/java/com/sun/findflight/ui/main/MainActivity.kt]: gộp lại thành `setBackButtonStatus(status:)` nhé

File: app/src/main/java/com/sun/findflight/ui/searchPlace/SearchPlaceFragment.kt [hunk 3]
```diff
@@ -1,32 +1,98 @@
...
+        registerForActivityResult(ActivityResultContracts.RequestPermission()) { isGranted ->
+            if (isGranted) {
+                showMessage(getString(R.string.text_permission_granted))
+            } else {
+                showMessage(getString(R.string.text_permission_denied))
+            }
+        }
+    private var presenter: SearchPlacePresenter? = null
+    private var placeListAdapter: PlaceListAdapter? =
+        PlaceListAdapter(this::itemPlaceClick, mutableListOf())
```
Comment (datdt-1935) [inline app/src/main/java/com/sun/findflight/ui/searchPlace/SearchPlaceFragment.kt]: dùng `by lazy` luôn em

File: app/src/main/java/com/sun/findflight/ui/searchPlace/SearchPlaceFragment.kt [hunk 4]
```diff
@@ -35,14 +101,93 @@ class SearchPlaceFragment : BaseFragment<FragmentSearchPlaceBinding>(), SearchPl
...
+    private fun getNearestPlaces(latitude: String, longitude: String) {
+        presenter?.getNearestPlaces(latitude, longitude)
+    }
+
     private fun itemPlaceClick(place: Place) {
+        setResult(place)
+    }
+
+    private fun setResult(place: Place) {
+        val bundle: Bundle = bundleOf(HomeFragment.KEY_DATA to place)
```
Comment (datdt-1935) [inline app/src/main/java/com/sun/findflight/ui/searchPlace/SearchPlaceFragment.kt]:
```suggestion
        val bundle = bundleOf(HomeFragment.KEY_DATA to place)
```

File: app/src/main/java/com/sun/findflight/ui/searchPlace/SearchPlaceFragment.kt [hunk 5]
```diff
@@ -35,14 +101,93 @@ class SearchPlaceFragment : BaseFragment<FragmentSearchPlaceBinding>(), SearchPl
...
+            else -> getPermission.launch(LOCATION_PERMISSION)
+        }
+    }
+
+    private fun closeKeyboard() {
+        activity?.currentFocus?.let {
+            val imm = activity?.getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
+            imm.hideSoftInputFromWindow(it.windowToken, FLAG_OPERATING)
+        }
+    }
```
Comment (datdt-1935) [inline app/src/main/java/com/sun/findflight/ui/searchPlace/SearchPlaceFragment.kt]: cái này có thể cân nhắc viết extension cho fragment

File: app/src/main/java/com/sun/findflight/ui/searchPlace/SearchPlacePresenter.kt [hunk 6]
```diff
@@ -1,12 +1,22 @@
...
+    override fun getNearestPlaces(latitude: String, longitude: String) {
+        view.showLoading()
     }
 
     override fun getData() {
 
     }
+
+    companion object {
+        const val ERROR_CANT_FIND_PLACE = "Can't find your place"
```
Comment (datdt-1935) [inline app/src/main/java/com/sun/findflight/ui/searchPlace/SearchPlacePresenter.kt]: để ra `strings.xml` nhé

Comment (vinhmone) [reply app/src/main/java/com/sun/findflight/ui/searchPlace/SearchPlacePresenter.kt]: cái này không có context thì lấy string ra kiểu gì nhỉ a @@

**Pull request merged**